### PR TITLE
Generate the main README.md from template and plugin documentation

### LINKED ISSUE #60: Improve how we manage the plugins README.md and the parent README.md
## TODO

Improve how we manage the plugins README.md and the parent README.md

## Description

Do you see a trick to merge the content of a README.md from subfolders into the parent README.md ?

Why ? To be able to see the README page of a backstage plugin: front, [console](https://github.com/q-shift/backstage-plugins/tree/main/plugins/quarkus-console) or backend (= subfolder of a git repository) on npmjs.org BUT also to see the whole documentation here on [main](https://github.com/q-shift/backstage-plugins/blob/main/README.md)

### PR DESCRIPTION
**Pull request by jacobdotcosta**
This GitHub workflow generates the root `README.md` file with the contents of the `README.m4.template.md`. The latter is a [GNU M4](https://www.gnu.org/software/m4/) template that includes each of the plugin `README.md` files at the end so their contents are added to the root `README.md`. The update is performed upon PR merge.

Closes #60

### CODE REVIEW

File: .github/workflows/pr-update-doc.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Generate main README.md from template
+        if: steps.pr_contains_readme.outputs.grep_result > 0
+        run: |
+          echo "Generate README.md from template"
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com"
+          m4 README.m4.template.md > README.md
+          git add README.md
+          git commit -m "Generated main README.md file from template"
```
Comment (cmoulliard) [inline .github/workflows/pr-update-doc.yaml:53]: I propose as text "Generate a new README.md merging content from plugins README"

File: README.m4.template.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+| Name                                                        | Description                                                                                                                     |
+|-------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------|
+| [QuarkusExtensionList](#Quarkus-extensions-field)           | Filter, select your Quarkus extensions using the `Quarkus Extension List` field.                                                |
+| [QuarkusQuickstartPicker](#Quarkus-Quickstart-picker-field) | Select using the `Quarkus QuickStart Picker` one of the quickstarts available: https://github.com/quarkusio/quarkus-quickstarts |
+| [QuarkusVersionList](#Quarkus-version-list-field)           | List the recommended and available versions of Quarkus                                                                          |
+
+**NOTE**: Such frontend feature(s) should be used with the quarkus scaffolder backend plugin (described hereafter) in order to get the generated project from https://code.quarkus.io/ as zip file !
+
+### Quarkus console
```
Comment (cmoulliard) [inline README.m4.template.md]: Content starting from here should be imported from the different plugins + README.md !!

File: .github/workflows/pr-update-doc.yaml (hunk 1 above)
Comment (jacobdotcosta) [reply .github/workflows/pr-update-doc.yaml:53]: Fixed. 

https://github.com/q-shift/backstage-plugins/pull/86/commits/b6fee0a8b4fa0c1d8d5da10e1b4bb9f947836b45#diff-33222b510ad7fadbf795f2b9f0732d86a5fc6c3a80c8c707a03fed5486fb51a5L46

File: README.m4.template.md (hunk 2 above)
Comment (jacobdotcosta) [reply README.m4.template.md]: Removed everything that comes after the Getting started section, while keeping that section.

https://github.com/q-shift/backstage-plugins/pull/86/commits/9e7e667bcfb6f1a851e5c87a3e4de3a042c4f0c1

**Comment by cmoulliard** [comment]
The template should include part of the main the following titles and renamed title:

```
# @qshift/quarkus-plugins

## Prerequisites

## Getting started

## Frontend plugin

### Quarkus console -> Console 

// import => plugins/quarkus-console

### Quarkus UI fields - > Scaffold template fields 

// import => plugins/quarkus

## Backend plugin

// import => plugins/quarkus-backend
```

Question: Should we import the README plugin file without or with its title ? 
I think that the README file imported should have its own title that we will include under either `## Frontend plugin` or `## Backend plugin`.

They should be renamed as what we by example developed under:
- `plugins/quarkus-backend` => corresponds to `Scaffold actions`
- `plugins/quarkus` => `Scaffold Template fields`
- `plugins/quarkus-console` => `Console` (aka Backstage entity tab)

Independently of this PR it will be needed that I update the content of each plugins/README as by example 
`plugins/quarkus-console/README.md` is completely wrong and outdated

**Comment by jacobdotcosta** [comment]
Since each of the plugins already have a 2nd level heading (`##`) as title I was assuming that they would be applied "AS IS". Not sure how to make transformation as it would mean excluding sections of the included plugin README files which might require changing them affecting how they look.

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**